protocol/lib/slices.go

```diff
@@ -16,17 +19,17 @@ func ContainsDuplicates[V comparable](values []V) bool {
 	return false
 }
 
-// ConvertMapToSliceOfKeys converts a map to a slice of keys.
-// Note that this method returns the keys in a non-deterministic order.
-func ConvertMapToSliceOfKeys[K comparable, V any](m map[K]V) []K {
-	var slice = make([]K, len(m))
-	index := 0
-	for key := range m {
-		slice[index] = key
-		index += 1
+// GetSortedKeys returns the keys of the map in sorted order.
+func GetSortedKeys[R interface {
```

Comment:
Should this be in `protocol/lib/maps` since it operates on a maps?

Reply:
We might just remove `lib/maps` (TBD) since a lot of those are primarily used for testing